### ISSUE TITLE
Punctuation mark in transcript

### ISSUE THREAD

**Issue by karan101292** [GoogleCloudPlatform/google-cloud-ruby#2079]
Hi, I was wondering if it is possible to get transcript with punctuation mark, as per their documentation it is possible and is still in beta phase (https://cloud.google.com/speech-to-text/docs/automatic-punctuation).

They have provided sample code for python but not for ruby. Is there any way we can try this in ruby?

I tried some syntax from python code reference but did not work.

**Comment by karan101292** [comment]
Hi @blowmage any idea on when we are planning to enable V1.1Beta1 API ?

**Comment by blowmage** [comment]
We have a big change coming to the gem in #1977. Those changes will have to be released before we can add V1.1beta1.

**Comment by jtoy** [comment]
@karan101292 anyway to get the relevant code?

**Comment by karan101292** [comment]
Hi @jtoy 
You can refer google speech API documentation for that.
here - https://cloud.google.com/speech-to-text/docs/reference/rest/v1p1beta1/speech/longrunningrecognize

first of all this API need to be authenticated. Only API key wont help you. You need to generate access_code for that using Google OAuth 2.0

I used `googleauth` gem for that (https://github.com/google/google-auth-library-ruby)

Code to generate access_token:
```
scope = "https://www.googleapis.com/auth/cloud-platform"
authorizer = Google::Auth::ServiceAccountCredentials.make_creds(json_key_io: File.open(ENV['GOOGLE_APPLICATION_CREDENTIALS']), scope: scope)
authorizer.fetch_access_token!
```

Code to perform speech-to-text:
```
url = URI.parse("https://speech.googleapis.com/v1p1beta1/speech:longrunningrecognize")
params = {
	"config": {
		"encoding":"FLAC",
		"languageCode": "en-US",
		"enableAutomaticPunctuation": true,
		"enableWordTimeOffsets": true
	},
	"audio": {
		"uri": uri
	}
}
req = Net::HTTP::Post.new(url.to_s)
req["Accept"] = "application/json"
req.content_type = "application/json"
req.body = params.to_json
req['Authorization'] = "Bearer #{authorizer.access_token}"
http = Net::HTTP.new(url.host, url.port)
http.use_ssl = (url.scheme == "https")
response = http.request(req)
response = response ? JSON.parse(response.body) : {}
```

In above request, API will send operation id in response with key - name:
something like - `{name: "5457221545124545"}`

You can check status of this operation using below API, this API doesn't require access_token, API key is enough.
```
operation_id =  "5457221545124545" 
url = URI.parse("https://speech.googleapis.com/v1/operations/#{operation_id}/?key=#{ENV['GOOGLE_API_KEY']}")
req = Net::HTTP::Get.new(url.to_s)
http = Net::HTTP.new(url.host, url.port)
http.use_ssl = (url.scheme == "https")
response = http.request(req)
response = response ? JSON.parse(response.body) : {}
```
here you will receive complete transcript in a response if operation is completed on google server.

I am using `LongRunningRecognizeResponse` as my audio file is big. If you are using small audio files, you can use ` POST https://speech.googleapis.com/v1p1beta1/speech:recognize`. This API will directly send transcript in a response instead of operation_id.

Hope it helps ;)

**Comment by blowmage** [comment]
We are still working to unblock this.

**Comment by brianarpie** [comment]
I'm looking forward to using beta features with this library!

**Comment by blowmage** [comment]
This is now available in [google-cloud-speech 0.30.0](https://github.com/GoogleCloudPlatform/google-cloud-ruby/releases/tag/google-cloud-speech%2Fv0.30.0). Be sure to specify the `V1p1beta1` API version to enable this. Also see the [migration guide](https://cloud.google.com/speech-to-text/docs/ruby-client-migration) for converting your existing code to the new library, as there are breaking changes.

**Comment by cigar2010** [comment]
the punctuation function is embeded into voice to text function as an option, is there any individual api or function processing un taged text with adding punctuation into text?
